### ISSUE TITLE
Question for testing kitti

### ISSUE THREAD

**Issue by SwagJ** [chrischoy/FCGF#20]
Dear Chris, 

I downloaded pre-trained model with normalized feature and dimension = 32. And when I ran test_kitti.py. It gives me  this. I set the "dataset" in config.json to be "KITTINMPairDataset". 
![Screenshot from 2020-02-28 21-20-13](https://user-images.githubusercontent.com/22762687/75584909-295d3500-5a71-11ea-86de-9321fc4bdc40.png)

Could you please tell me how to fix this problem? Thank you very much.

**Comment by chrischoy** [comment]
The script requires config.json, which should contain correct paths to weights, conv1_kernel_size, whether normalize features, and model_n_out.

Could you make sure that you are loading correct weights and if so, correct configuration?

**Comment by SwagJ** [comment]
Dear Chris, 

In my config.json: conv_kernel_size=5 and normalize features  = true,  model_n_out = 32,  which should match the last one on your released trained model. And after I downloaded it, and load into model ResUNetBN2C, the loading is successful under my local configuration. Display the network as follow. 

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Model's state_dict:
conv1.kernel 	 torch.Size([125, 1, 32])
norm1.bn.weight 	 torch.Size([32])
norm1.bn.bias 	 torch.Size([32])
norm1.bn.running_mean 	 torch.Size([32])
norm1.bn.running_var 	 torch.Size([32])
norm1.bn.num_batches_tracked 	 torch.Size([])
block1.conv1.kernel 	 torch.Size([27, 32, 32])
block1.norm1.bn.weight 	 torch.Size([32])
block1.norm1.bn.bias 	 torch.Size([32])
block1.norm1.bn.running_mean 	 torch.Size([32])
block1.norm1.bn.running_var 	 torch.Size([32])
block1.norm1.bn.num_batches_tracked 	 torch.Size([])
block1.conv2.kernel 	 torch.Size([27, 32, 32])
block1.norm2.bn.weight 	 torch.Size([32])
block1.norm2.bn.bias 	 torch.Size([32])
block1.norm2.bn.running_mean 	 torch.Size([32])
block1.norm2.bn.running_var 	 torch.Size([32])
block1.norm2.bn.num_batches_tracked 	 torch.Size([])
conv2.kernel 	 torch.Size([27, 32, 64])
norm2.bn.weight 	 torch.Size([64])
norm2.bn.bias 	 torch.Size([64])
norm2.bn.running_mean 	 torch.Size([64])
norm2.bn.running_var 	 torch.Size([64])
norm2.bn.num_batches_tracked 	 torch.Size([])
block2.conv1.kernel 	 torch.Size([27, 64, 64])
block2.norm1.bn.weight 	 torch.Size([64])
block2.norm1.bn.bias 	 torch.Size([64])
block2.norm1.bn.running_mean 	 torch.Size([64])
block2.norm1.bn.running_var 	 torch.Size([64])
block2.norm1.bn.num_batches_tracked 	 torch.Size([])
block2.conv2.kernel 	 torch.Size([27, 64, 64])
block2.norm2.bn.weight 	 torch.Size([64])
block2.norm2.bn.bias 	 torch.Size([64])
block2.norm2.bn.running_mean 	 torch.Size([64])
block2.norm2.bn.running_var 	 torch.Size([64])
block2.norm2.bn.num_batches_tracked 	 torch.Size([])
conv3.kernel 	 torch.Size([27, 64, 128])
norm3.bn.weight 	 torch.Size([128])
norm3.bn.bias 	 torch.Size([128])
norm3.bn.running_mean 	 torch.Size([128])
norm3.bn.running_var 	 torch.Size([128])
norm3.bn.num_batches_tracked 	 torch.Size([])
block3.conv1.kernel 	 torch.Size([27, 128, 128])
block3.norm1.bn.weight 	 torch.Size([128])
block3.norm1.bn.bias 	 torch.Size([128])
block3.norm1.bn.running_mean 	 torch.Size([128])
block3.norm1.bn.running_var 	 torch.Size([128])
block3.norm1.bn.num_batches_tracked 	 torch.Size([])
block3.conv2.kernel 	 torch.Size([27, 128, 128])
block3.norm2.bn.weight 	 torch.Size([128])
block3.norm2.bn.bias 	 torch.Size([128])
block3.norm2.bn.running_mean 	 torch.Size([128])
block3.norm2.bn.running_var 	 torch.Size([128])
block3.norm2.bn.num_batches_tracked 	 torch.Size([])
conv4.kernel 	 torch.Size([27, 128, 256])
norm4.bn.weight 	 torch.Size([256])
norm4.bn.bias 	 torch.Size([256])
norm4.bn.running_mean 	 torch.Size([256])
norm4.bn.running_var 	 torch.Size([256])
norm4.bn.num_batches_tracked 	 torch.Size([])
block4.conv1.kernel 	 torch.Size([27, 256, 256])
block4.norm1.bn.weight 	 torch.Size([256])
block4.norm1.bn.bias 	 torch.Size([256])
block4.norm1.bn.running_mean 	 torch.Size([256])
block4.norm1.bn.running_var 	 torch.Size([256])
block4.norm1.bn.num_batches_tracked 	 torch.Size([])
block4.conv2.kernel 	 torch.Size([27, 256, 256])
block4.norm2.bn.weight 	 torch.Size([256])
block4.norm2.bn.bias 	 torch.Size([256])
block4.norm2.bn.running_mean 	 torch.Size([256])
block4.norm2.bn.running_var 	 torch.Size([256])
block4.norm2.bn.num_batches_tracked 	 torch.Size([])
conv4_tr.kernel 	 torch.Size([27, 256, 128])
norm4_tr.bn.weight 	 torch.Size([128])
norm4_tr.bn.bias 	 torch.Size([128])
norm4_tr.bn.running_mean 	 torch.Size([128])
norm4_tr.bn.running_var 	 torch.Size([128])
norm4_tr.bn.num_batches_tracked 	 torch.Size([])
block4_tr.conv1.kernel 	 torch.Size([27, 128, 128])
block4_tr.norm1.bn.weight 	 torch.Size([128])
block4_tr.norm1.bn.bias 	 torch.Size([128])
block4_tr.norm1.bn.running_mean 	 torch.Size([128])
block4_tr.norm1.bn.running_var 	 torch.Size([128])
block4_tr.norm1.bn.num_batches_tracked 	 torch.Size([])
block4_tr.conv2.kernel 	 torch.Size([27, 128, 128])
block4_tr.norm2.bn.weight 	 torch.Size([128])
block4_tr.norm2.bn.bias 	 torch.Size([128])
block4_tr.norm2.bn.running_mean 	 torch.Size([128])
block4_tr.norm2.bn.running_var 	 torch.Size([128])
block4_tr.norm2.bn.num_batches_tracked 	 torch.Size([])
conv3_tr.kernel 	 torch.Size([27, 256, 64])
norm3_tr.bn.weight 	 torch.Size([64])
norm3_tr.bn.bias 	 torch.Size([64])
norm3_tr.bn.running_mean 	 torch.Size([64])
norm3_tr.bn.running_var 	 torch.Size([64])
norm3_tr.bn.num_batches_tracked 	 torch.Size([])
block3_tr.conv1.kernel 	 torch.Size([27, 64, 64])
block3_tr.norm1.bn.weight 	 torch.Size([64])
block3_tr.norm1.bn.bias 	 torch.Size([64])
block3_tr.norm1.bn.running_mean 	 torch.Size([64])
block3_tr.norm1.bn.running_var 	 torch.Size([64])
block3_tr.norm1.bn.num_batches_tracked 	 torch.Size([])
block3_tr.conv2.kernel 	 torch.Size([27, 64, 64])
block3_tr.norm2.bn.weight 	 torch.Size([64])
block3_tr.norm2.bn.bias 	 torch.Size([64])
block3_tr.norm2.bn.running_mean 	 torch.Size([64])
block3_tr.norm2.bn.running_var 	 torch.Size([64])
block3_tr.norm2.bn.num_batches_tracked 	 torch.Size([])
conv2_tr.kernel 	 torch.Size([27, 128, 64])
norm2_tr.bn.weight 	 torch.Size([64])
norm2_tr.bn.bias 	 torch.Size([64])
norm2_tr.bn.running_mean 	 torch.Size([64])
norm2_tr.bn.running_var 	 torch.Size([64])
norm2_tr.bn.num_batches_tracked 	 torch.Size([])
block2_tr.conv1.kernel 	 torch.Size([27, 64, 64])
block2_tr.norm1.bn.weight 	 torch.Size([64])
block2_tr.norm1.bn.bias 	 torch.Size([64])
block2_tr.norm1.bn.running_mean 	 torch.Size([64])
block2_tr.norm1.bn.running_var 	 torch.Size([64])
block2_tr.norm1.bn.num_batches_tracked 	 torch.Size([])
block2_tr.conv2.kernel 	 torch.Size([27, 64, 64])
block2_tr.norm2.bn.weight 	 torch.Size([64])
block2_tr.norm2.bn.bias 	 torch.Size([64])
block2_tr.norm2.bn.running_mean 	 torch.Size([64])
block2_tr.norm2.bn.running_var 	 torch.Size([64])
block2_tr.norm2.bn.num_batches_tracked 	 torch.Size([])
conv1_tr.kernel 	 torch.Size([96, 64])
final.kernel 	 torch.Size([64, 32])
final.bias 	 torch.Size([1, **32])**

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
So I am somehow get confused. I think it should be right? I also attach my config.json file(it's in .txt for upload, but locally it's .json file). I think the data is loaded correctly. Otherwise, script will throw assertion error. 

[config.txt](https://github.com/chrischoy/FCGF/files/4270481/config.txt)

**Comment by chrischoy** [comment]
Your config.txt has null for weights. The script will test KITTI dataset with randomly initialized weights. Please populate the weights field with an appropriate path to the downloaded weights.

**Comment by SwagJ** [comment]
Dear Chris, 

Thanks for your help. Problem was out_dir in config.json instead of null weights. I think weight is only needed for training purpose, although it is null as well in config.json for 3DMatch training. Anyway, thank you very much for helping. 

Also, I have an additional question. Have you test FCGF on large-scale scene datasets like KAIST? I was wondering about FCGF's performance on large-scale scene dataset.

**Comment by chrischoy** [comment]
Glad that you solved it.

I am not familiar with the KAIST dataset. Could you let me know the full name of the dataset?

**Comment by SwagJ** [comment]
KAIST Urban Dataset is the full name. It is developed by KAIST and NAVER labs.  Here's the link:  https://irap.kaist.ac.kr/dataset/index.html

**Comment by chrischoy** [comment]
The dataset looks very impressive. I don't think I'll get back to the 3D feature learning problem soon, but will keep you posted if I do evaluate on the dataset.

**Comment by SwagJ** [comment]
I see. Well, Thank you very much for your work. It's very impressive.
